### ISSUE TITLE
charset=UTF-8 does not work

### ISSUE THREAD

**Issue by ygra** [magjac/graphviz-visual-editor#187]
There's a sample linked from the documentation:

http://magjac.com/graphviz-visual-editor/?dot=digraph%20G%20%7b%0a%20%20charset%3d%22UTF-8%22%0a%20%20%f0%9f%8d%94%20-%3e%20%f0%9f%92%a9%0a%7d

which immediately raises an error, even though as part of the official documentation, that should probably work.

**Comment by magjac** [comment]
I agree this is a bug. I wonder if it's possible that this is an [upstream bug](https://gitlab.com/graphviz/graphviz/-/issues/1687) which is now fixed. I'll attempt an upgrade.

**Comment by ygra** [comment]
Judging from the parser source code, could it be that code points outside the BMP aren't even supported? The parser looks at individual UTF-16 code units and doesn't appear to consider surrogates at all.

**Comment by magjac** [comment]
This *is* a bug in the Graphviz Visual Editor itself, not an upstream bug.

A workaround is to quote the node names [like this](http://magjac.com/graphviz-visual-editor/?dot=digraph%20G%20%7B%0A%20%20charset%3D%22UTF-8%22%0A%20%20%22%F0%9F%8D%94%22%20-%3E%20%22%F0%9F%92%A9%22%0A%7D):

```
digraph G {
  charset="UTF-8"
  "🍔" -> "💩"
}
```
